### PR TITLE
feat(pie-modal): DSW-2339 recombine props for leading and supporting actions

### PR DESCRIPTION
**Pull request by xander-marjoram**
## Describe your changes (can list changeset entries if preferable)
- Recombine `leadingActionText`, `leadingActionVariant` and `aria.leadingActionLabel` into a single `leadingAction` prop.
- Recombine `supportingActionText`, `supportingActionVariant` and `aria.supportingActionLabel` into a single `supportingAction` prop.

Four modal visual tests were removed because other tests offer sufficient coverage:
- "Modal displays footer" - other tests cover showing the buttons, and the footer only renders if there are buttons.
- "Modal displays modal footer" - same as above
- "Modal does not display footer" - covered by tests which check that the buttons don't render
- "Modal falls back to default property `primary`" - the tests no longer pass in the `variant` if using the default value, so other tests now cover this automatically.

One more visual test added:
- "Modal leadingAction variant can be overridden"

## Author Checklist (complete before requesting a review)
- [x] I have performed a self-review of my code
- [x] I have added thorough tests where applicable (unit / component / visual)
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] I have reviewed visual test updates properly before approving
- [x] If changes will affect consumers of the package, I have created a changeset entry.
- [x] If a changeset file has been created, I have used the `/snapit` functionality to test my changes in a consuming application

### Reviewer 1 - @jamieomaguire 
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] If there are visual test updates, I have reviewed them

### Reviewer 2 @leksaBoiko 
- [x] I have reviewed the `PIE Storybook`/`PIE Docs` PR preview
- [x] If there are visual test updates, I have reviewed them

### CODE REVIEW

File: packages/components/pie-modal/src/index.ts [hunk 1]
```diff
@@ -307,26 +301,26 @@ export class PieModal extends RtlMixin(LitElement) implements ModalProps {
...
-     * The (optional) aria-label is read from the `aria` prop's `leadingActionLabel` property.
+     * If `leadingAction.text` is not provided, the button is not rendered.
+     * If `leadingAction.variant` is not provided, the default value of "primary" is used.
+     * The (optional) aria-label is read from `leadingAction.ariaLabel`.
      *
      * @private
      */
     private renderLeadingAction () : TemplateResult | typeof nothing {
-        if (!this.leadingActionText) {
+        if (!this.leadingAction?.text) {
```
Comment (jamieomaguire) [inline packages/components/pie-modal/src/index.ts:311]: Entirely personal preference but I wonder if restructuring the `text`, `variant` and `ariaLabel` properties from the object could be cleaner here. Would make the template render less wordy. Up to you!

**Review by jamieomaguire — DISMISSED** [review]

**Comment by leksaBoiko** [comment]
apps/pie-docs/src/components/modal/code/props.json - `aria` description, it seems that `leadingAction` and `supportingAction` are no longer needed here: 

"An object representing the aria labels for the `close`, `back`, `leadingAction` and `supportingAction` buttons within the modal as well as the `isLoading` state labels (`aria-label` & `aria-busy`)."

**Comment by leksaBoiko** [comment]
just to clarify, will you create a follow-up PR to update `pie-modal` props in `pie-aperture`?

**Review by leksaBoiko — DISMISSED** [review]

**Review by jamieomaguire — APPROVED** [review]

**Pull request merged**